mopidy-mpris: init at 3.0.1  mopidy-somafm: init at 2.0.0  mopidy-youtube: 2.0.2 -> 3.0  mopidy-gmusic: 3.0.0 -> 4.0.0

NickHu opened the pull request:
`mopidy-gmusic: 3.0.0 -> 4.0.0`
`mopidy-mpris: init at 3.0.1`
`mopidy-somafm: init at 2.0.0`
`mopidy-youtube: 2.0.2 -> 3.0.0`

<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change

~~This changeset moves mopidy onto the python3 branch (the python2 version is EOL), and updates some plugins. Note that `mopidy-mpd` was bundled with mopidy previously, but has been extracted into its own plugin in this release.~~ This was merged in a different PR.

~~This PR is slightly WIP as in I cannot figure out how to get `mopidy-youtube` to build; any help debugging this build error would be appreciated.~~ Fixed by @balsoft's comment below.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

balsoft commented:
I'm really happy to see this, have been waiting for somebody to do that!

This works for beautifulsoup:
```
{ stdenv, python3Packages, mopidy }:

python3Packages.buildPythonApplication rec {
  pname = "mopidy-youtube";
  version = "3.0a1";
  
  src = python3Packages.fetchPypi {
    inherit version;
    pname = "Mopidy-YouTube";
    sha256 = "111b70x4z9ym5578jz3c327cj4l1vkfczgr4y3ggvapa5x2r7maf";
  };
  
  patchPhase = "sed s/bs4/beautifulsoup4/ -i setup.cfg";
  
  propagatedBuildInputs = [
    mopidy
    python3Packages.beautifulsoup4
    python3Packages.youtube-dl
    python3Packages.cachetools
  ];
  
  doCheck = false;
  
  meta = with stdenv.lib; {
    description = "Mopidy extension for playing music from YouTube";
    license = licenses.asl20;
    maintainers = [ maintainers.spwhitt ];
  };
}
```
So my uneducated guess is that there is a problem with the way beautifulsoup is packaged. Alternatively, we can submit an issue upstream asking whether `bs4` is really the correct dependency name.

balsoft commented:
@GrahamcOfBorg build mopidy mopidy-gmusic mopidy-youtube mopidy-mpd mopidy-mpris mopidy-somafm

balsoft commented:
Also I believe it's better to have a separate commit for each package.

jonringer reviewed (dismissed):
correct, each package addition and bump should be a separate commit, so 7 in total

NickHu committed:
mopidy-gmusic: 3.0.0 -> 4.0.0

NickHu committed:
mopidy-mpris: init at 3.0.1

NickHu committed:
mopidy-somafm: init at 2.0.0

NickHu committed:
mopidy-youtube: 2.0.2 -> 3.0

jonringer reviewed (commented):
please take a look at package failures:

```
[15 built (6 failed), 14 copied (20.9 MiB), 4.6 MiB DL]
error: build of '/nix/store/gznv1kjf542cb5dv02dl5ayx3jffwjyf-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/78361
7 package failed to build:
mopidy-moped mopidy-mopify mopidy-musicbox-webclient mopidy-soundcloud mopidy-spotify mopidy-spotify-tunigo mopidy-youtube

8 package built:
mopidy mopidy-gmusic mopidy-mpd mopidy-mpris mopidy-somafm python27Packages.pykka python37Packages.pykka python38Packages.pykka
```

most have similar error:
```
  Processing ./Mopidy_Spotify-3.1.0-py2.py3-none-any.whl
  ERROR: Could not find a version that satisfies the requirement Mopidy>=2.0 (from Mopidy-Spotify==3.1.0) (from versions: none)
  ERROR: No matching distribution found for Mopidy>=2.0 (from Mopidy-Spotify==3.1.0)
```

NickHu commented:
Right, these are the mopidy plugins which I didn't update (except `mopidy-youtube` - I can't for the life of me figure out why that's not building) - a lot of them have actually become obsolete, in the sense that they do not support the latest (python3) version of mopidy (see https://mopidy.com/ext/). Should I just remove them or would you suggest another approach? Bearing in mind python2 is now officially considered EOL now

jonringer commented:
unfortunately, you cannot have different versions of the same package in `python-modules`, you could try bumping that which is still compatibly with v2.0.

The other option would be to disable the other packages, however, I don't like the idea of adding packages that are already broken.

NickHu commented:
I think you might be misunderstanding me a little - mopidy has migrated to python3, and as such it does not support python2 plugins any more. The plugins which still work have been rewritten/migrated to python3. But many plugins have not been migrated, and I think that we should just remove them

FRidh commented:
needs a rebase

jonringer commented:
> I think you might be misunderstanding me a little - mopidy has migrated to python3, and as such it does not support python2 plugins any more. The plugins which still work have been rewritten/migrated to python3. But many plugins have not been migrated, and I think that we should just remove them

If there's no plan to migrate the broken plugins, then please remove. Otherwise you can just mark them broken until they are in a usable state. I would just really like for package evaluation reflect the current condition of the package

jpotier commented:
At least spotify-tunigo can be marked as broken without too much thinking: upstream project is deprecated, and most features are now implemented in the mopidy-spotify plugin.

balsoft commented:
I can't figure out how to get `mopidy gmusic login` to work. Does anybody have any idea?

NickHu commented:
Sorry, I really let this one wither a bit - I updated, rebased, and trimmed off the bits that got merged into other PRs, so I think this ought to be good to merge now

NickHu commented:
@balsoft I just set up a new instance of mopidy, and did a `mopidy gmusic login` and it worked fine. Note that there's an unfortunate bug https://github.com/mopidy/mopidy-gmusic/issues/231 which causes no tracks to appear when trying to select an album/artist (e.g. in ncmpcpp, pressing "2", Google Play Music -> Albums -> etc), but playing tracks from the All tracks view still works. It was fixed between the last release and master; you can use the following overlay to use the version currently on master of mopidy-gmusic instead:

```
self: super:
{
  mopidy-gmusic = super.mopidy-gmusic.overrideAttrs (oldAttrs: rec {
    src = self.fetchFromGitHub {
      owner = "mopidy";
      repo = "mopidy-gmusic";
      rev = "a4a78463b9765af62dd8662d45c2f8d9718f2d63";
      sha256 = "116h7v35vy0pvkab0diwfyrf8lcpbam9can8vx5cyjqxhazigzj7";
    };
  });
}
```

NickHu commented:
@GrahamcOfBorg build mopidy-gmusic mopidy-youtube mopidy-mpris mopidy-somafm

balsoft commented:
@NickHu oh, now it works for me too. I guess I have somehow messed up something last time I tried it ;)

NickHu commented:
@balsoft @jpotier @jonringer @FRidh Any remaining objections or should I merge this?

balsoft commented:
Can't wait for it to be merged, one more hacky overlay removed from my config!

Thank you very much for this once again!

NickHu commented:
Result of `nixpkgs-review pr 78361` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
<br>- mopidy-mpris
<br>- mopidy-somafm
<br>- mopidy-youtube
</details>
